File: e-learning-admin/src/models/course.ts
import { CourseStatus } from "@/enums/courseStatus";

export interface CourseReq {
    name: string;
    description: string;
    audience: string;
    target: string;
    content: string;
    thumbnail: File | string;
    price: number;
    status: CourseStatus;
    categoryId: number;
    levelId: number;
}

export interface CourseShortRes {
    id: number;
    name: string;
    thumbnail: string;
    status: CourseStatus;
    categoryName: string;
    levelName: string;
}

export interface CourseRes {
    id: number;
    name: string;
    description: string;
    audience: string;//
    target: string;//
    content: string;//
    thumbnail: string;
    price: number;
    status: CourseStatus;
    categoryId: number;
    levelId: number;
}

File: e-learning-api/src/main/java/com/example/elearningapi/entity/Course.java
package com.example.elearningapi.entity;

import com.example.elearningapi.enums.CourseStatus;
import jakarta.persistence.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Entity
@Table(name = "courses")
public class Course {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @Column(columnDefinition = "LONGTEXT")
    private String description;

    @Column(columnDefinition = "TEXT")
    private String audience;

    @Column(columnDefinition = "TEXT")
    private String target;

    @Column(columnDefinition = "LONGTEXT")
    private String content;

    @Column(columnDefinition = "LONGTEXT")
    private String thumbnail;

    private Double price;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private CourseStatus status;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    @ManyToOne
    @JoinColumn(name = "level_id")
    private Level level;

    @OneToMany(mappedBy = "course", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Lesson> lessons;

    @OneToMany(mappedBy = "course", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<UserCourse> userCourses;
}
